jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletChannel.java

```diff
@@ -781,14 +782,27 @@ void onTrailers(HttpFields trailers)
 
     public void onCompleted()
     {
-        HttpServletRequest httpServletRequest = _request.getHttpServletRequest();
+        ServletContextRequest.ServletApiRequest apiRequest = _request.getServletApiRequest();
         if (LOG.isDebugEnabled())
-            LOG.debug("onCompleted for {} written={}", httpServletRequest.getRequestURI(), getBytesWritten());
+            LOG.debug("onCompleted for {} written={}", apiRequest.getRequestURI(), getBytesWritten());
 
         long idleTO = _configuration.getIdleTimeout();
         if (idleTO >= 0 && getIdleTimeout() != _oldIdleTimeout)
             setIdleTimeout(_oldIdleTimeout);
 
+        if (getServer().getRequestLog() != null)
+        {
+            Authentication authentication = apiRequest.getAuthentication();
```

Comment:
Can we add a TODO here for only adding the attributes if the custom request logger has these things in the format string?   Better yet, in this PR, implement that check in the doStart method and have an EnumSet for if user name; real path; handlerName are needed, then only set the attributes if those booleans are set.